pa11ycrawler/cmdline.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/python
+"""
+The command line interface for pa11ycrawler.
+"""
+import argparse
+
+from textwrap import dedent
+
+from scrapy.crawler import CrawlerProcess
+
+from pa11ycrawler.reporter import HtmlReporter
+from pa11ycrawler.settings_helpers import (
+    is_pa11y_setting,
+    is_pa11ycrawler_setting,
+    get_default_settings,
+)
+from pa11ycrawler.spiders.pa11y_spider import Pa11ySpider
+
+# ------------ Main CLI ------------
+
+
+def cli():
+    """
+    Returns the main CLI parser.
+    """
+    parser = argparse.ArgumentParser(prog='pa11ycrawler')
+    subparsers = parser.add_subparsers(title='commands')
+    run_cli(subparsers)
+    json_to_html_cli(subparsers)
+    return parser
+
+# ------------ "run" CLI -----------
+
+
+def run_cli(subparsers):
+    """
+    Creates the parser for the `run` command
+    """
+    run_usage = dedent('''pa11ycrawler run [options] url [url ...]
+
+    ---------------------------------------------------------------
+    NOTE: This only documents a haldful of the possible options.
+    More documentation can be found for the following categories:
+
+    CUSTOM SCRAPY SETTINGS:
+    ** Implemented for pa11ycrawler. (e.g. Not found in Scrapy Docs.)
+    ** All options documented below.
+    ** Most likely to need to be changed.
```

Comment:
Ah, no, this is for the user.   It could definitely be worded better.  The point that I was trying to make is that these are the settings that will likely need to be set when running the command.  For example, one of the settings is `PA11YCRAWLER_ALLOWED_DOMAINS`, which you'd most likely want to set to limit the crawl to specifics domain.

Maybe this line is excessive.